dashboard/views.py

```diff
@@ -143,6 +145,86 @@ def post(self, request):
         )
 
 
+class UserExamEnrollment(APIView):
+    """
+    Create an exam enrollment for the user in a given exam run identified by course_id.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def post(self, request):
+        """
+        enrolls the user in a exam run in edx
+        """
+        edx_exam_course_id = request.data.get('exam_course_id')
+
+        if edx_exam_course_id is None:
+            raise ValidationError('exam course id missing in the request')
+
+        exam_run = ExamRun.objects.get(edx_exam_course_key=edx_exam_course_id)
```

Comment:
This would be a bit cleaner if the `exam_course_id` was part of the url, and I would just do a simple `get_object_or_404` on it.